### PR TITLE
Encourage better PR descriptions

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

This repo contains a PR template to help contributors understand how to fill out the description for a new PR. The template is intended more for reference, however, and so it does not go into detail on the differences between the Description and Changes sections. This leads to PR descriptions that lack the detail that the Shared Libraries team would prefer in order to
ensure that pull requests can be reviewed more thoroughly and that changelogs can be compiled more quickly and contain more high-quality information after a release is issued.

This commit attempts to improve pull request descriptions going forward by adding a new section to the Contributing document. It also updates the pull request template to:

- Rename Description to Explanation so as not to overload the term "description"
- Rename Changes to Changelog so as not to overload the term "changes"
- Move up References, as it's most commonly provided after the explanation
- Update both Explanation and Changelog to include the same details mentioned in Contributing
- Update Changelog to ensure that changes are divided by package name

## Changelog

(None)

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -18,6 +18,39 @@
 >
 > `workspaceName` in these commands is the `name` field within a package's `package.json`, e.g., `@metamask/address-book-controller`, not the directory where it is located, e.g., `packages/address-book-controller`.
```
Comment (Gudahtt) [inline docs/contributing.md:19]: Nit: Is this note now redundant?

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: docs/contributing.md (hunk 1 above)
Comment (mcmire) [reply docs/contributing.md:19]: Hmm, do you mean that we shouldn't need to explain what a workspace name is now?

Comment (Gudahtt) [reply docs/contributing.md:19]: I thought it was a repeat of what you added below on line 36. Though now I see that it's not exactly a repeat (it's making the same clarification but regarding two different subjects), so you can disregard this comment.

**Pull request merged**